### PR TITLE
NH-65069 Lambda layer 0.8 MB decrease

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
This shrinks APM Python lambda layer zip size from 9.5 MB to 8.7 MB by removing super-upstream deps that don't seem to be needed for actual instrumentation. Otel Python does something similar in its builds: https://github.com/open-telemetry/opentelemetry-lambda/blob/main/python/src/otel/Dockerfile#L19-L20

Also removes an unused target `publish-lambda-layer-rc` and changes the temp dir name used for lambda publish, as it was conflicting with [this dir](https://github.com/solarwindscloud/solarwinds-apm-python/tree/main/tmp/solarwinds-apm) for cert stuff.

Tested that instrumentation still works by merging this with [Settings API PoC](https://github.com/solarwindscloud/solarwinds-apm-python/pull/207) branch, building and uploading new [layer](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/layers/tammy_test_apm_python_m1_buildtest/versions/7?tab=versions), attaching layer to [function](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/tammy_test_python_m1_buildtest?tab=monitoring), and invoking function. Traces and metrics export still work:
* https://my.na-02.cloud.solarwinds.com/158363224169902080/entities/services/e-1656498477873647616/traces/requests?duration=300&sort_by=time&sort_order=descending&heatmap_type=spanDuration&endTime=1699411800
* https://my.na-02.cloud.solarwinds.com/158363224169902080/entities/services/e-1656498477873647616/service-metrics?duration=300&endTime=1699411800

### CODE REVIEW

**Comment by tammy-baylis-swi** [comment]
Ready for review. Not sure why reviewers didn't automatically add when I first put this in.

File: Makefile [hunk 1]
```diff
@@ -185,13 +185,13 @@ package: sdist check-sdist-local manylinux-wheels check-wheel-local
...
 # (x86_64 OR aarch64)
-target_dir := "./tmp"
+target_dir := "./tmp-lambda"
 aws-lambda: check-zip wrapper
 	@if [ -f ./dist/solarwinds_apm_lambda_${platform}.zip ]; then \
 		echo -e "Deleting old solarwinds_apm_lambda_${platform}.zip"; \
 		rm ./dist/solarwinds_apm_lambda_${platform}.zip; \
 	 fi
-	rm -rf ./tmp
+	rm -rf ./tmp-lambda
```
Comment (cheempz) [inline Makefile:194]: Minor curiosity why this and a few other commands below don't use `${target_dir}` instead?

**Review by cheempz — APPROVED** [review]
LGTM @tammy-baylis-swi!

File: Makefile (hunk 1 above)
Comment (tammy-baylis-swi) [reply Makefile]: No reason! Fixed in [29e600c](https://github.com/solarwinds/apm-python/pull/214/commits/29e600c63bef8d313117eff458c74a9e98c9dd18) 😸

**Comment by tammy-baylis-swi** [comment]
Thank you @cheempz ! One more look when you have a moment please.

**Review by cheempz — APPROVED** [review]
👍 thanks for the revisit!

**Pull request merged**